src/librustc_mir/transform/generator.rs

```diff
@@ -484,18 +483,119 @@ fn locals_live_across_suspend_points(
             // and their storage is live (the `storage_liveness` variable)
             storage_liveness.intersect(&liveness.outs[block]);
 
+            // The generator argument is ignored
+            storage_liveness.remove(self_arg());
+
             let live_locals = storage_liveness;
```

Comment:
This rename should be done before `storage_liveness.intersect(&liveness.outs[block]);`